ang/taskRunner.js

```diff
@@ -61,53 +61,63 @@
         }
       }
 
-      if ($scope.errors.length > 0 ) {
-        return;
-      }
+      if ($scope.errors.length > 0 ) return;
 
       const data = {
         task_id: taskId,
-        async: true,
+        async: await $scope.isBackgroundQueueEnabled(),
       };
 
       if ($scope.isInputValueRequired()) {
         data['input_val'] = $scope.inputValue;
       }
 
       $scope.isShowLogs = true;
```

Comment:
Previously, `isShowLogs` was only set after the `Sqltask.execute` call is done. The log display logic here has changed now, but since this property is also used to determine if the `Task "%1" has been executed. [...]` message is displayed in `taskRunner.html`, the UX is now a bit confusing, showing this message immediately after task execution starts, possibly making the user believe execution has finished.

I would suggest leaving the `Are you sure you want to execute "%1"?` message as-is (with `ng-if="!isShowLogs"`), and then using a separate property to track the task execution status (success/error). The message could then be either `Task "%1" has been executed successfully. [...]` or `Task "%1" has encountered an error. [...]`, ideally with some sort of error styling.

While the task is running (in terms of properties: `isShowLogs == true` and the new execution status is empty or something like that), we could show "Task is executing ...", maybe with different styling, to indicate ongoing execution.